src/couch_changes.erl

```diff
@@ -349,6 +366,19 @@ check_docids(_) ->
     throw({bad_request, Msg}).
 
 
+check_selector(Selector={_}) ->
+    try
+        mango_selector:normalize(Selector)
+    catch
+        {mango_error, _Mod, Reason} -> % see MANGO_ERROR(R) in mango.hrl
+            Msg = io_lib:format("Selector error: ~p", [Reason]),
+            throw({bad_request, Msg})
+    end;
+check_selector(Selector) ->
+    Msg = io_lib:format("Selector error: ~p, expected an object", [Selector]),
```

Comment:
May be not send back the Selector data? The case: send ~1GiB of data like `{"selector": [....]}` and let server try to serialize that 1GiB of data and send it back to you.